### PR TITLE
Swipe listener feature

### PR DESCRIPTION
**Pull request by StevenHaa**
swiping left will go to video, swiping right will go to photo. There is a visual queue to let the user know which mode he/she is on

### CODE REVIEW

**Comment by StevenHaa** [comment]
i have pulled from your branch @younesstawil into our branch so your changes/commits should now appear in the PR and resolved

**Comment by Earthii** [comment]
Pull @younesstawil branch for the fix for #86

**Comment by Earthii** [comment]
Added the unit tests to this PR from #93

File: app/src/main/kotlin/com/simplemobiletools/camera/activities/MainActivity.kt [hunk 1]
```diff
@@ -676,7 +692,21 @@ class MainActivity : SimpleActivity(), PreviewListener, PhotoProcessor.MediaSave
         fadeAnim(last_photo_video_preview, 1f)
         scheduleFadeOut()
     }
+    fun fadePhotoVideo(){
```
Comment (n2chao) [inline app/src/main/kotlin/com/simplemobiletools/camera/activities/MainActivity.kt]: What do these values mean in the fadeAnim?

File: app/src/main/kotlin/com/simplemobiletools/camera/views/Preview.kt [hunk 2]
```diff
@@ -100,6 +104,31 @@ class Preview : ViewGroup, SurfaceHolder.Callback, MediaScannerConnection.OnScan
...
+            }
+            override fun onSwipeRight(){
+                //go to photo mode
+                if(swipeIndex == 1){
+                    mActivity!!.swipeBetweenPhotoVideo()
+                    mActivity!!.fadePhotoVideo()
+                    swipeIndex--
+                }
+            }
+            override fun onSwipeTop(){
```
[the remark this answers is not in the document]
Comment (StevenHaa) [reply app/src/main/kotlin/com/simplemobiletools/camera/views/Preview.kt]: For now no, but maybe in the future swiping down can go to settings, and swiping up can show filters/preview photos? These are just ideas, nothing is final

File: app/src/main/kotlin/com/simplemobiletools/camera/activities/MainActivity.kt (hunk 1 above)
Comment (StevenHaa) [reply app/src/main/kotlin/com/simplemobiletools/camera/activities/MainActivity.kt]: The value ranges from 0-1, the higher the value, the darker the outline. So "0" means its invisible/transparent and 1 means its totally visible. 0.7 mean its visible but more transparent than the 1. I'll add this into the code

File: app/src/main/kotlin/com/simplemobiletools/camera/views/Preview.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (n2chao) [reply app/src/main/kotlin/com/simplemobiletools/camera/views/Preview.kt]: I see, thanks for clearing that out, Steven.

File: app/src/main/kotlin/com/simplemobiletools/camera/activities/MainActivity.kt (hunk 1 above)
Comment (n2chao) [reply app/src/main/kotlin/com/simplemobiletools/camera/activities/MainActivity.kt]: Alright, I see. Thanks.

**Pull request merged**